aws/ecs/atlantis.tf

```diff
@@ -50,10 +50,124 @@ variable "atlantis_container_memory" {
+  description = "Cognito User Pool ARN"
+  default     = ""
+}
+
+variable "atlantis_cognito_user_pool_client_id" {
+  type        = "string"
+  description = "Cognito User Pool Client ID"
+  default     = ""
+}
+
+variable "atlantis_cognito_user_pool_domain" {
+  type        = "string"
+  description = "Cognito User Pool Domain. The User Pool Domain should be set to the domain prefix (`xxx`) instead of full domain (https://xxx.auth.us-west-2.amazoncognito.com)"
+  default     = ""
+}
+
+variable "atlantis_google_oidc_client_id" {
+  type        = "string"
+  description = "Google OIDC Client ID. Use this URL to create a Google OAuth 2.0 Client and obtain the Client ID and Client Secret: https://console.developers.google.com/apis/credentials"
+  default     = ""
+}
+
+variable "atlantis_google_oidc_client_secret" {
+  type        = "string"
+  description = "Google OIDC Client Secret. Use this URL to create a Google OAuth 2.0 Client and obtain the Client ID and Client Secret: https://console.developers.google.com/apis/credentials"
+  default     = ""
+}
+
+variable "atlantis_alb_ingress_listener_unauthenticated_priority" {
+  type        = "string"
+  default     = "50"
+  description = "The priority for the rules without authentication, between 1 and 50000 (1 being highest priority). Must be different from `alb_ingress_listener_authenticated_priority` since a listener can't have multiple rules with the same priority"
+}
+
+variable "atlantis_alb_ingress_listener_authenticated_priority" {
+  type        = "string"
+  default     = "100"
+  description = "The priority for the rules with authentication, between 1 and 50000 (1 being highest priority). Must be different from `alb_ingress_listener_unauthenticated_priority` since a listener can't have multiple rules with the same priority"
+}
+
+variable "atlantis_alb_ingress_unauthenticated_paths" {
+  type        = "list"
+  default     = ["/events"]
+  description = "Unauthenticated path pattern to match (a maximum of 1 can be defined)"
+}
+
+variable "atlantis_alb_ingress_authenticated_paths" {
+  type        = "list"
+  default     = ["/*"]
+  description = "Authenticated path pattern to match (a maximum of 1 can be defined)"
+}
+
+locals {
+  authentication_enabled = "${var.atlantis_authentication_action_type == "COGNITO" || var.atlantis_authentication_action_type == "GOOGLE_OIDC" ? "true" : "false"}"
+
+  supported_authentication_actions = {
+    "COGNITO" = {
+      type = "authenticate-cognito"
+
+      authenticate_cognito = [{
+        user_pool_arn       = "${var.atlantis_cognito_user_pool_arn}"
+        user_pool_client_id = "${var.atlantis_cognito_user_pool_client_id}"
+        user_pool_domain    = "${var.atlantis_cognito_user_pool_domain}"
+      }]
+    }
+
+    "GOOGLE_OIDC" = {
+      type = "authenticate-oidc"
+
+      authenticate_oidc = [{
+        client_id     = "${var.atlantis_google_oidc_client_id}"
+        client_secret = "${var.atlantis_google_oidc_client_secret}"
```

Comment:
I thought the idea was to send the SSM parameter key for the secret and have the module retrieve it.